pydantic/schema.py

```diff
@@ -643,7 +645,8 @@ def add_field_type_to_schema(field_type: Any, schema: Dict[str, Any]) -> None:
     and then modifies the given `schema` with the information from that type.
     """
     for type_, t_schema in field_class_to_schema:
-        if issubclass(field_type, type_):
+        # Fallback for `typing.Pattern` as it is not a valid class
+        if lenient_issubclass(field_type, type_) or field_type is type_ is Pattern:
```

Comment:
We may have a better check later to support for example `Pattern[str]` but `typing` API changed so much that we'll probably first need some utils to retrieve the right informations depending on python version.
I feel like this is a good enough solution for the time being

Reply:
https://github.com/samuelcolvin/pydantic/issues/2399 also hits this
a python >= 3.9 solution is in https://github.com/samuelcolvin/pydantic/pull/2392